nixos/modules/services/misc/paperless.nix

```diff
@@ -199,9 +207,41 @@ in
       defaultText = literalExpression "pkgs.paperless-ngx";
       description = lib.mdDoc "The Paperless package to use.";
     };
+
+    backup = {
+
+      enable = mkEnableOption (lib.mdDoc "Paperless backups");
+
+      directory = mkOption {
+        type = types.str;
+        default = cfg.dataDir + "/backup";
+        description = lib.mdDoc "Directory to store backups.";
+      };
+
+      calendar = mkOption {
+        type = types.str;
+        default = "01:30:00";
+        description = lib.mdDoc ''
+          Configured when to run the backup service systemd unit (DayOfWeek Year-Month-Day Hour:Minute:Second).
+        '';
+      };
+
+      options = mkOption {
+         type = types.str;
+         default = "-c -d";
+         description = lib.mdDoc "Options to pass to the document exporter.";
+      };
+
+      postScript = mkOption {
```

Comment:
A preScript could also be useful. It should run after vars like destination and soruce are set but before the exporter.

I hardlink the media to the export dir first such that the export takes no additional space for example.

See https://github.com/Atemu/nixos-config/blob/acb974f94de9ab3a16928a9bb01051a1cf1ec2f6/paperless.nix#L49-L70